cmd/shell.go

```diff
@@ -137,8 +148,40 @@ func StatsAndLogging(logConf SyslogConfig) (metrics.Scope, blog.Logger) {
 	cfsslLog.SetLogger(cfsslLogger{logger})
 	_ = mysql.SetLogger(mysqlLogger{logger})
 	grpclog.SetLogger(grpcLogger{logger})
+	return logger
+}
 
-	return scope, logger
+func makeStats(addr string, logger blog.Logger) metrics.Scope {
+	registry := prometheus.NewRegistry()
+	registry.MustRegister(prometheus.NewGoCollector())
+	registry.MustRegister(prometheus.NewProcessCollector(os.Getpid(), "boulder"))
+
+	mux := http.NewServeMux()
+	reg := func(name string) {
+		mux.Handle("/debug/pprof/"+name, pprof.Handler(name))
+	}
+	reg("block")
```

Comment:
Could you add a comment for what this is doing? Appears slightly magic.

Also nit: this seems like something that could replaced with a `for _, stat := range []string{"goroutine", "heap", ...} { reg(stat) }`.

Reply:
I wound up not doing the for loop, since I think it doesn't make things clearer or easier to read. The line-by-line repetition is pretty straightforward and short.